File: quickbytes_backend/src/v1/models/article.ts
import { Prisma } from "@prisma/client";
import { NewsCategory, Relevancy } from "./news_category";

export interface Article {
  id?: string;
  title: string;
  content: string;
  image: string;
  published_on: Date;
  categories: string[] | Prisma.article$categoriesArgs;
  source_url: string;
  user_id: string;
}

File: android/app/src/main/kotlin/com/quickbytes/quickbytes_app/Models/Article.kt
import com.quickbytes.quickbytes_app.Models.NewsCategory
import kotlinx.serialization.*

@Serializable
data class Article (
    val id: String,
    val title: String,
    val content: String,
    val image: String,

    @SerialName("published_on")
    val publishedOn: String,

    @SerialName("categories")
    val categories: List<NewsCategory>,

    @SerialName("source_url")
    val sourceURL: String
)
